src/coreclr/pal/src/include/pal/virtual.h

```diff
@@ -191,6 +193,10 @@ class ExecutableMemoryAllocator
     // will try to reserve during initialization. We want all JIT-ed code and the
     // entire libcoreclr to be located in a 2GB range.
     static const int32_t MaxExecutableMemorySize = 0x7FFF0000;
+#else
+    static const int32_t CoreClrLibrarySize = 16 * 1024 * 1024;
```

Comment:
We can use this smaller value for x64 too (I was actually testing it with 10MB). The original value is bogus and seems to be incorrectly based on the size of the binary including the debug symbols.

Reply:
Changed to 10mb

Reply:
Correct me if I am wrong but a bad guessing is not really a problem - it just most likely will cause the first "reserve" probe to fail - we'll append 4mb to the address for the next probe 🤷‍♂️

Reply:
I think if you underestimate the library size then `MaxExecutableMemorySizeNearCoreClr` will be too large and the very first allocation we do might not be able to reach the end of libcoreclr.so on x64.

Reply:
@EgorBo the x64 Debug build is actually 13953648 large, so the 16MB you had there was actually a good estimate. Unless the arm64 debug build is larger.